src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -209,6 +252,158 @@ def ne(self, y):
 
         """
 
+    @staticmethod
+    def reconstruct_semi_honest(
+        share_ptrs: List["ReplicatedSharedTensor"],
+        get_shares: bool = False,
+    ) -> Union[torch.Tensor, List[torch.Tensor]]:
+        """Reconstruct value from shares.
+
+        Args:
+            share_ptrs (List[RSTensorPointers]): List of RSTensor pointers.
+            get_shares (bool): Retrieve only shares.
+
+        Returns:
+            reconstructed_value (torch.Tensor): Reconstructed value.
+        """
+        shares1 = share_ptrs[0].get_shares()[0].get()
+        shares2 = share_ptrs[1].get_shares().get()
+
+        shares = []
+
+        if isinstance(shares2, torch.Tensor):
+
+            shares = [shares1] + shares2
+
+        else:
+
+            shares = [shares1] + shares2
+
+        if get_shares:
+            return shares
+
+        return sum(shares)
+
+    @staticmethod
+    def reconstruct_malicious(
```

Comment:
Could you have only a reconstruct method and depending on the protocol type you call ```reconstruct_malicious``` or ```reconstruct_semi_honest``` (I am also thinking that this reconstruct malicious/semi-honest should live somewhere in the protocol, but let's keep it here for the moment).

Also, since ```reconstruct_semi_honest/malicious``` would be accessed only internally you can have them with a ```_```(underscore) as a prefix

Reply:
hmm we can have a parameter in reconstructing for RST, for the type of reconstruction

Reply:
**LE**: Add only the ```_``` to the method.
I did not see the ```reconstruct```